modules/ui/arrow-ui-data/src/test/kotlin/arrow/ui/SumTest.kt

```diff
@@ -39,17 +42,29 @@ class SumTest : UnitSpec() {
     val IDEQ = Eq<Kind<ForId, Int>> { a, b -> Id.eq(Int.eq()).run { a.fix().eqv(b.fix()) } }
     val IDH = Hash<Kind<ForId, Int>> { Id.hash(Int.hash()).run { it.fix().hash() } }
 
+    val cf2: (Int) -> Sum<Kind<ForConst, Int>, Kind<ForConst, Int>, Int> = { Sum.left(Const.just(it), Const.just(it)) }
+    val g2 = Gen.int().map(cf2) as Gen<Kind<SumPartialOf<ConstPartialOf<Int>, ConstPartialOf<Int>>, Int>>
+
     testLaws(
       DivisibleLaws.laws(
         Sum.divisible(Const.divisible(Int.monoid()), Const.divisible(Int.monoid())),
-        { Sum.left(Const.just(it), Const.just(it)) },
+        g2,
         Eq { a, b ->
-          a.fix().side == b.fix().side &&
-            a.fix().left.value() == b.fix().left.value() &&
-            a.fix().right.value() == b.fix().right.value()
+          (a.fix() to b.fix()).let {
+            when (it.first.side) {
+              is Sum.Side.Left -> when (it.second.side) {
+                is Sum.Side.Left -> it.first.left.value() == it.second.left.value()
+                else -> false
+              }
+              is Sum.Side.Right -> when (it.second.side) {
+                is Sum.Side.Right -> it.first.right.value() == it.second.right.value()
+                else -> false
+              }
+            }
+          }
```

Comment:
Moving all of this to `EqK` would also be nice

Reply:
that was really interesting. in the generalized solution i had to use to eqk's inside the sum eqk.

Reply:
Yup, that will happen a few times more with mtl gens :+1: